action_server/src/action_server/actions/find.py

```diff
@@ -49,68 +48,132 @@ def _point_at_person(self, person):
                 else:
-                    self._config_result.message = " Where should I look for the {}?".format(self._object.type)
+                    self._config_result.message = " Where should I look for the {}?".format(self._semantics.object.type)
                 self._config_result.missing_field = 'source-location'
                 return
-            self._location = EntityDescription(id=e.id)
         else:
-            self._config_result.message = " Where should I look for the {}?".format(self._object.id)
+            self._config_result.message = " Where should I look for the {}?".format(self._semantics.object.type)
             self._config_result.missing_field = 'source-location'
             return
 
-        # We (safely) assume that self._location is an EntityDescription object
-        # If we need to find a manipulable item, the location should also be manipulable
-        if not self._object.type == "person" and self._location.id not in self._knowledge.manipulation_locations \
-            and self._location.id not in self._knowledge.location_rooms:
-            self._config_result.message = " I can't grasp anything from the {}".format(self._location.id)
+        # We can now assume we know the source location
+        # Set up context about the found entity
+        self._config_result.context['object'] = config.semantics['object']
+
+        # Check if we need to find a person or an object
+        if self._semantics.object.type == 'person':
+            # If no source-location is specified, check if a location is specified in the object spec
+            if not self._semantics.source_location:
+                if self._semantics.object.location:
+                    self._semantics.source_location = self._semantics.object.location
+                elif self._context.location:
+                    self._semantics.source_location = self._context.location
+                else:
+                    self._config_result.message = " Where should I look for {}?".format(self._semantics.object.id)
+                    self._config_result.missing_field = "source-location"
+                    return
+
+            # if we need to find the person near a furniture object, get the room where it is located
+            if self._semantics.source_location.id not in self._knowledge.location_rooms:
+                self._semantics.source_location.id = self._knowledge.get_room(self._semantics.source_location.id)
+
+            # person in room
+            if not self._semantics.object.id:
+                self._semantics.object.id = 'someone'
+            self._found_entity_designator = VariableDesignator(resolve_type=Entity)
+            self._find_state_machines = [
+                states.FindPersonInRoom(robot, self._semantics.source_location.id, self._semantics.object.id,
+                                        True, self._found_entity_designator.writeable)]
```

Comment:
The fourth parameter should be variable depending on if the robot is looking for:

-a known person -> True
-Just anyone       -> False

Can be done in the same way as done in line 130